### PR TITLE
Added whiteListedWorlds config option

### PR DESCRIPTION
**Pull request by sh0inx**

### CODE REVIEW

File: src/main/java/com/iridium/iridiumteams/managers/MissionManager.java [hunk 1]
```diff
@@ -43,22 +43,25 @@ public LocalDateTime getExpirationTime(MissionType missionType, LocalDateTime st
...
-     * @param environment The environment we are in
+     * @param missionWorld       The world we are in
      * @param missionType The mission type e.g. BREAK
      * @param identifier  The mission identifier e.g. COBBLESTONE
      * @param amount      The amount we are incrementing by
      */
-    public void handleMissionUpdate(T team, World.Environment environment, String missionType, String identifier, int amount) {
+    public void handleMissionUpdate(T team, World missionWorld, String missionType, String identifier, int amount) {
+
+        if (iridiumTeams.getConfiguration().whitelistedWorlds.stream().noneMatch(world -> missionWorld.getName().equalsIgnoreCase(world))) return;
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumteams/managers/MissionManager.java:53]: We probabbly also want to make it so if the list is empty, then it allows any world

File: src/main/java/com/iridium/iridiumteams/configs/Configuration.java [hunk 2]
```diff
@@ -117,5 +118,6 @@ public Configuration(String color, String team, String pluginName) {
                 )), Collections.emptyList(), 0, new HashMap<>(), 2000, 0, XSound.ENTITY_PLAYER_LEVELUP))
                 .build();
 
+        this.whitelistedWorlds = Arrays.asList("world");
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumteams/configs/Configuration.java:121]: This should be an empty list by default imo

File: src/test/java/com/iridium/iridiumteams/managers/MissionManagerTest.java
Comment (PeachesMLG) [inline src/test/java/com/iridium/iridiumteams/managers/MissionManagerTest.java:1]: Can we add a test for when they do a mission in a world thats not on the list, and assert that the mission doesnt progress?

handleMissionUpdate_ShouldNotIncrementValue_WhenMissionIsNotInWhitelistedWorld

**Review by PeachesMLG — DISMISSED** [review]

File: src/main/java/com/iridium/iridiumteams/managers/MissionManager.java (hunk 1 above)
Comment (sh0inx) [reply src/main/java/com/iridium/iridiumteams/managers/MissionManager.java:53]: oh thats a good idea, i will do that

File: src/test/java/com/iridium/iridiumteams/managers/MissionManagerTest.java
Comment (sh0inx) [reply src/test/java/com/iridium/iridiumteams/managers/MissionManagerTest.java:1]: ya sure

**Pull request merged**